### PR TITLE
Insights: migrate code insights from repo-updater to worker service

### LINKED ISSUE #21531: insights: Isolate the code insights background jobs from repo-updater
Currently a number of background [jobs](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/enterprise/internal/insights/background/background.go#L28) are started for the code insights backend that currently live within the scope of the repo-updater service.

Code insights should migrate to a worker model ( #16197) to reduce the operational impact on other services. This is also necessary to facilitate future horizontal scaling needs for the code insights backend.

### PR DESCRIPTION
**Pull request by coury-clark**
Closes #21531 

Migrate code insights backend jobs from repo-updater to worker service.

This migration is pretty naive and only aims to remove the repo-updater dependency. We will split jobs up based on horizontal scalability in a future PR.

### CODE REVIEW

**Comment by coury-clark** [comment]
Still need to update changelog after it's cut for 3.30

File: enterprise/cmd/worker/internal/insights/job.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+)
+
+type insightsBaseConfig struct {
+	env.BaseConfig
+
+	enabled bool
+}
+
+func (i *insightsBaseConfig) Load() {
+	log15.Info("loading config")
```
Comment (coury-clark) [inline enterprise/cmd/worker/internal/insights/job.go:21]:
```suggestion
```

File: enterprise/cmd/worker/internal/insights/job.go [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+	i.enabled = insights.IsEnabled()
+	log15.Info("enabled value", "value", i.enabled)
```
Comment (coury-clark) [inline enterprise/cmd/worker/internal/insights/job.go:23]:
```suggestion
```

File: internal/goroutine/periodic.go [hunk 3]
```diff
@@ -75,12 +75,14 @@ func (h *simpleHandler) HandleError(err error) {
...
-// NewPeriodicGoroutine creates a new PeriodicGoroutine with the given handler.
+// NewPeriodicGoroutine creates a new PeriodicGoroutine with the given handler. The context provided will propagate into
+// the executing goroutine and will terminate the goroutine if cancelled.
 func NewPeriodicGoroutine(ctx context.Context, interval time.Duration, handler Handler) *PeriodicGoroutine {
 	return NewPeriodicGoroutineWithMetrics(ctx, interval, handler, nil)
 }
 
-// NewPeriodicGoroutine creates a new PeriodicGoroutine with the given handler.
+// NewPeriodicGoroutineWithMetrics creates a new PeriodicGoroutine with the given handler. The context provided will propagate into
+// // the executing goroutine and will terminate the goroutine if cancelled.
```
Comment (efritz) [inline internal/goroutine/periodic.go:85]: Strange formatting here.

**Review by efritz — APPROVED** [review]

File: enterprise/internal/insights/background/background.go [hunk 4]
```diff
@@ -27,26 +25,11 @@ import (
 	"github.com/sourcegraph/sourcegraph/internal/workerutil/dbworker"
 )
 
-// StartBackgroundJobs is the main entrypoint which starts background jobs for code insights. It is
+// GetBackgroundJobs is the main entrypoint which starts background jobs for code insights. It is
 // called from the repo-updater service, currently.
```
Comment (coury-clark) [inline enterprise/internal/insights/background/background.go:29]:
```suggestion
// called from the worker service.
```

File: internal/goroutine/periodic.go (hunk 3 above)
Comment (coury-clark) [inline internal/goroutine/periodic.go:85]:
```suggestion
// the executing goroutine and will terminate the goroutine if cancelled.
```

**Pull request merged**